### DISCUSSION TITLE
Short for `vgld`?

### DISCUSSION THREAD

**Discussion by txtyash** [helix-editor/helix#863] [Q&A] [answered]
Vim: D (deletes till the end of the line).
hx: ?

**Comment by pickfire** [comment, marked as answer]
`vgld`, `v` for starting the extend mode, `gl` for going to end of line, `d` for delete.

Yeah, it's a lot of keys and if you know a good and consistent idea to solve it, feel free to suggest.

**Reply by kajuburfi** [reply]
Can't we do `xd`?
`x`: Select line
`d`: Delete

**Reply by NNBnh** [reply]
But this delete the whole line not "deletes till the end of the line" (deletes from the current cursor position to EOL)

**Reply by pickfire** [reply]
`xd` deletes the current line but what he wants is delete till end of line.

I wish `gl` can select till end of line by default but that would make it inconsistent with the rest of `g?` keys.

**Reply by ath3** [reply]
if someone wants to have it the vim way, put this in helix config:

[keys.normal]
D = ["select_mode", "goto_line_end", "delete_selection"]

**Reply by txtyash** [reply]
Why is this mapping not available by default?

**Reply by txtyash** [reply]
I got this error after using that mapping and now helix won't even open:
```
> hx ~/.config/helix/config.toml
Error: data did not match any variant of untagged enum KeyTrie for key `keys.normal` at line 14 column 1
```

**Reply by ath3** [reply]
You need to build helix from git to use it.

**Reply by txtyash** [reply]
Oh okay.

**Reply by pickfire** [reply]
I believe maybe your distro helix is a bit old, helix have quite active development recently and version was updated a few times in the past months.

> Why is this mapping not available by default?

Because it is very specific and we probably don't want to special case this only. Unless everyone find it useful.
